rust/src/ortc.rs

```diff
@@ -77,32 +78,36 @@ impl RtpMapping {
         }
     }
 
-    pub(crate) fn from_fbs(mapping: rtp_parameters::RtpMapping) -> Self {
-        Self {
+    pub(crate) fn from_fbs_ref(
+        mapping: rtp_parameters::RtpMappingRef<'_>,
+    ) -> Result<Self, Box<dyn Error>> {
+        Ok(Self {
             codecs: mapping
-                .codecs
+                .codecs()?
                 .iter()
-                .map(|mapping| RtpMappingCodec {
-                    payload_type: mapping.payload_type,
-                    mapped_payload_type: mapping.mapped_payload_type,
+                .map(|mapping| {
+                    Ok(RtpMappingCodec {
+                        payload_type: mapping?.payload_type()?,
+                        mapped_payload_type: mapping?.mapped_payload_type()?,
+                    })
                 })
-                .collect(),
+                .collect::<Result<Vec<_>, Box<dyn Error>>>()?,
             encodings: mapping
-                .encodings
+                .encodings()?
                 .iter()
-                .map(|mapping| RtpMappingEncoding {
-                    rid: mapping.rid.clone().map(|rid| rid.to_string()),
-                    ssrc: mapping.ssrc,
-                    scalability_mode: mapping
-                        .scalability_mode
-                        .clone()
-                        .unwrap_or(String::from("S1T1"))
-                        .parse()
-                        .unwrap(),
-                    mapped_ssrc: mapping.mapped_ssrc,
+                .map(|mapping| {
+                    Ok(RtpMappingEncoding {
+                        rid: mapping?.rid()?.map(|rid| rid.to_string()),
+                        ssrc: mapping?.ssrc()?,
+                        scalability_mode: mapping?
+                            .scalability_mode()?
+                            .unwrap_or(String::from("S1T1").as_str())
```

Comment:
I'd say you should parse it first and then use `unwrap_or_default()` rather than using a string here (I know it is not the change of this PR).

Reply:
`scalability_mode()` is of type `Result<Option<String>>`. I think it would not apply in this case, right?

Reply:
Why not? `.map(|maybe_scalability_mode| maybe_scalability_mode.parse()).transpose()?.unwrap_or_default()`

You map `Option<String>` into `Option<Result<ScalabilityMode>, Error>`, then turn it into `Result<Option<ScalabilityMode>, Error>`, bubble up result and default to default value if it is still `None`.